core/src/main/java/feign/Request.java

```diff
@@ -303,6 +306,33 @@ public static class Options {
     private final long readTimeout;
     private final TimeUnit readTimeoutUnit;
     private final boolean followRedirects;
+    private final Map<String, Map<String, Options>> threadToMethodOptions;
+
+    /**
+     * Get an Options by methodName
+     * 
+     * @param methodName it's your FeignInterface method name.
+     * @return method Options
+     */
+    public Options getMethodOptions(String methodName) {
```

Comment:
My spider sense is telling me this will cause headaches at some point.

Since I don't have a good suggestions for right now, just annotate and the new public methods with `@Experimental` and I will merge.  Just want to reserve the right to undo/redo in the future.

Reply:
Ok, added.

Reply:
Missing `@Experimental` on this one =)